File: dns_client/dns_client.py
import sys
import socket

if __name__ == '__main__':

  if len(sys.argv) != 2:
    sys.stderr.write("usage: dns_client address\n")
    exit(1)

  hostname = sys.argv[1]
  ip_address = socket.gethostbyname(hostname)
  print("IP address: {0}".format(ip_address))

File: dns_client/dns_client.cs
using System;
using System.Linq;
using System.Net;
using static System.Net.Dns;

namespace DnsClient
{
    class Program
    {
        static void Main(string[] args)
        {
            // c# 7.0 tinkering, use this version of input error handling if you want
            if (args.Length != 1)
            {
                Console.Error.WriteLine("Usage: run with parameter <host-name>");
                Environment.Exit(1);
            }
            var hostname = args[0];
            try
            {
                IPHostEntry entry = GetHostEntry(hostname); //GetHostByName method is obsolete
                var ip = entry.AddressList.FirstOrDefault();
                Console.WriteLine($"Resolved IP address: {ip}");
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }
    }
}
